vignettes/sensitivity-analysis.Rmd

```diff
@@ -0,0 +1,118 @@
+sim <- loadSimulation(simFilePath)
+
+# Create a default `SensitivityAnalysis` for the simulation
+sa <- SensitivityAnalysis$new(simulation = sim)
+print(sa)
+
+# Create a `SensitivityAnalysis` with specified parameters
+sa <- SensitivityAnalysis$new(simulation = sim, parameterPaths = c("Aciclovir|Lipophilicity",
+                                                                      "Aciclovir|Fraction unbound (plasma)",
+                                                                      "Organism|Age"))
+print(sa)
+
+# Show which parameters will be variated
+sa$parameterPaths
+```
+
+New parameters can be added to an existing `SensitivityAnalysis` by calling the method `addParameterPaths`. WARNING: If no parameters were specified during creating of a `SensitivityAnalysis` (all constant parameters are considered), calling `addParameterPaths` will make only the manually added parameters being variated.
+
+```{r addPathsToSensitivityAnalysis}
+library(ospsuite)
+
+# Load simulation
+simFilePath <- file.path(getwd(), "..", "tests", "data", "Aciclovir.pkml", fsep = .Platform$file.sep)
+sim <- loadSimulation(simFilePath)
+
+# Create a `SensitivityAnalysis` with specified parameters
+sa <- SensitivityAnalysis$new(simulation = sim, parameterPaths = c("Aciclovir|Lipophilicity",
+                                                                      "Aciclovir|Fraction unbound (plasma)",
+                                                                      "Organism|Age"))
+# Add new parameter
+sa$addParameterPaths("Organism|Weight")
+
+# Show which parameters will be variated
+sa$parameterPaths
+```
+
+To run the defined specified `SensitivityAnalysis`, call the method `runSensitivityAnalysis`. The method returns an object of the class `SensitivityAnalysisResults`. The impact of the defined parameters is calculated for PK-Parameters (see [PK Analysis](pk-analysis.html) for more information) of all model outputs.
+
+```{r runSA}
+# Load simulation
+simFilePath <- file.path(getwd(), "..", "tests", "data", "Aciclovir.pkml", fsep = .Platform$file.sep)
+sim <- loadSimulation(simFilePath)
+
+# Create a `SensitivityAnalysis` with all constant parameters
+sa <- SensitivityAnalysis$new(simulation = sim)
+
+# Run the sensitivity analysis
+saResult <- runSensitivityAnalysis(sa)
+print(saResult)
+```
+The method `allPKParameterSensitivitiesFor` returns a list of `PKParameterSensitivity` objets. `PKParameterSensitivity` describes the sensitivity (field `$value`) of a PK-Parameter (`$pkParameterName`) for the output `$outputPath` calculated for the variated parameter `$parameterPath`. The argument `totalSensitivityThreshold` of the method `allPKParameterSensitivitiesFor` is used to filter out the most impactful parameters. A threshold of 0.9 means that only parameters participating to a total of 90 percent of the sensitivity would be returned. A value of 1 would return the sensitivity for all parameters.
+
```

Comment:
Is the threshold optional or mandatory in R? If optional: you could mention that 0.9 is also the **default** value

Reply:
The threshold is optional. I would prefer implementing a method to read the variables of the ospsuiteEnv, where such defaults are defined, as proposed here https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/386

Then we could put a referenrece to it in the documentation.

Reply:
I agree I think we need this